includes/fbbot/class-omise-fbbot-payment-handler.php

```diff
@@ -0,0 +1,253 @@
+
+      wp_enqueue_script( 'omise-js', 'https://cdn.omise.co/omise.js', array( 'jquery' ), OMISE_WOOCOMMERCE_PLUGIN_VERSION, true );
+    
+      wp_enqueue_script( 'omise-util', plugins_url( 'omise-woocommerce/assets/javascripts/omise-util.js' ), array( 'omise-js' ), OMISE_WOOCOMMERCE_PLUGIN_VERSION, true );
+
+      wp_enqueue_script( 'omise-payment-on-messenger-form-handler', plugins_url( 'omise-woocommerce/assets/javascripts/omise-payment-on-messenger-form-handler.js'), array( 'jquery' ), OMISE_WOOCOMMERCE_PLUGIN_VERSION, true);
+
+      wp_localize_script( 'omise-payment-on-messenger-form-handler', 'omise_params', array(
+        'key'       => $public_key,
+        'vault_url' => OMISE_VAULT_HOST
+      ) );
+    }
+
+    add_action( 'wp_enqueue_scripts', 'load_scripts' );
+	}
+
+  function set_oms_key() {
+    define( 'OMISE_PUBLIC_KEY', $this->public_key );
+    define( 'OMISE_SECRET_KEY', $this->private_key );
+  }
+
+	public static function get_instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	public function parameter_queryvars( $qvars ) {
+		$qvars[] = 'product_id';
+		$qvars[] = 'messenger_id';
+    $qvars[] = 'error_message';
+
+		return $qvars;
+	}
+
+	public function payment_page_detect( $posts ) {
+		global $wp;
+    global $wp_query;
+		global $payment_page_detect;
+    
+    $payment_page_url = "pay-on-messenger";
+    $payment_purchase_complete = "complete-payment";
+    $payment_error_url = "pay-on-messenger-error";
+
+    if ( strtolower( $wp->request ) == $payment_purchase_complete ) {
+      $image_url = site_url() . '/wp-content/plugins/omise-woocommerce/assets/images/omise_logo.png';
+      $url = 'https://www.messenger.com/closeWindow/?image_url=' . $image_url . '&display_text=THANKS%20FOR%20PURCHASE';
+      if ( wp_redirect( $url ) ) {
+        exit;
+      }
+
+      return $posts;
+    }
+
+    // Create custom page
+    $post = new stdClass;
+
+    if ( strtolower( $wp->request ) == $payment_page_url ) {
+      $post->post_title = __( 'Your order' );
+      $post->post_content = $this->payment_page_render();
+    }
+
+    if ( strtolower( $wp->request ) == $payment_error_url ) {
+      $post->post_title = __( 'System error' );
+      $post->post_content = $this->payment_error_page_render();
+    }
+
+    if ( strtolower( $wp->request ) == $payment_page_url || strtolower( $wp->request ) == $payment_error_url ) {
+      // Create cumtom page content
+      $post->post_author = 1;
+      $post->post_name = strtolower( $wp->request );
+      $post->guid = get_bloginfo( 'wpurl' ) . '/' . strtolower( $wp->request );
+      $post->ID = -1;
+      $post->post_type = 'page';
+      $post->post_status = 'static';
+      $post->comment_status = 'closed';
+      $post->ping_status = 'open';
+      $post->comment_count = 0;
+      $post->post_date = current_time( 'mysql' );
+      $post->post_date_gmt = current_time( 'mysql', 1 );
+      $posts = NULL;
```

Comment:
@igroomgrim this should be an `array` instead of `null` (because of the next line)?